recipes/flac/all/conanfile.py

```diff
@@ -0,0 +1,70 @@
+from conans import ConanFile, tools, CMake
+import os
+
+
+class FlacConan(ConanFile):
+    name = "flac"
+    description = "Free Lossless Audio Codec"
+    topics = ("conan", "flac", "codec", "audio", )
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/xiph/flac"
+    license = ("BSD", "GPL", "LPGL")
+    requires = "ogg/1.3.4"
+    exports_sources = "CMakeLists.txt",
+    generators = "cmake",
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+    _source_subfolder = "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def build_requirements(self):
+        self.build_requires("nasm/2.14")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "{}-{}".format(self.name, self.version)
+        os.rename(extracted_dir, self._source_subfolder)
+        tools.replace_in_file(
+            os.path.join(self._source_subfolder, 'src', 'libFLAC', 'CMakeLists.txt'),
+            'target_link_libraries(FLAC PRIVATE $<$<BOOL:${HAVE_LROUND}>:m>)',
+            'target_link_libraries(FLAC PUBLIC $<$<BOOL:${HAVE_LROUND}>:m>)')
+        tools.replace_in_file(
+            os.path.join(self._source_subfolder, 'CMakeLists.txt'),
+            'set(CMAKE_EXE_LINKER_FLAGS -no-pie)',
+            '#set(CMAKE_EXE_LINKER_FLAGS -no-pie)')
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["BUILD_EXAMPLES"] = False
+        cmake.definitions["BUILD_DOCS"] = False
+        cmake.definitions["BUILD_TESTING"] = False
+        cmake.configure()
+        return cmake
```

Comment:
This avoids reconfiguration between `build` and `package`
```suggestion
        if self._cmake:
            return self._cmake
        self._cmake = CMake(self)
        self._cmake.definitions["BUILD_EXAMPLES"] = False
        self._cmake.definitions["BUILD_DOCS"] = False
        self._cmake.definitions["BUILD_TESTING"] = False
        self._cmake.configure()
        return self._cmake
```